### ISSUE TITLE
Performance for this function should be O(n)

### ISSUE THREAD

**Issue by antonio081014** [raywenderlich/swift-algorithm-club#893]
https://github.com/raywenderlich/swift-algorithm-club/blob/81bc0705cac3c1154a5bf248954034d7d532f2cb/Heap/Heap.swift#L215

In this function, index(of:) costs O(n), then remove(at:) costs O(log n), it should be O(n + log n), which would be O(n)

**Comment by antonio081014** [comment]
https://github.com/raywenderlich/swift-algorithm-club/blob/81bc0705cac3c1154a5bf248954034d7d532f2cb/Heap/Heap.swift#L215:L221

**Comment by antonio081014** [comment]
```
  /** Removes the first occurrence of a node from the heap. Performance: O(n log n). */
  @discardableResult public mutating func remove(node: T) -> T? {
    if let index = index(of: node) {
      return remove(at: index)
    }
    return nil
  }
```

**Comment by kelvinlauKL** [comment]
Fixed #905. Thanks!
